wordpress-mobile/WordPress-iOS
# Menus: Refactor Menus Create/Update Methods

Fixes #5433

As discussed in https://github.com/wordpress-mobile/WordPress-iOS/pull/5424#discussion_r64437740 a refactor was needed to clean up how Menus are created and updated.

This PR combines the create and update methods into one method `createOrUpdateMenu:` on `MenusService` that handles the creation, if needed, and updates the menu there-after.

Test Changes:

1. Open Menus on a .com site with admin rights.
2. Make changes to an existing Menu and Save.
3. Ensure the menu changes are reflected on the site as expected.

Test Create:

1. Create a new Menu by hitting the Menus selector in the header.
2. Edit the default "New Item" or add new items.
3. Save the new menu.
4. Ensure the new menu is reflected on the site as expected.

Needs review: @aerych would you mind taking a look since we originally discussed this one?
